### PR TITLE
Add user mapping to insecure authenticator

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/server/security/InsecureAuthenticatorConfig.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ */
+package io.prestosql.server.security;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.validation.FileExists;
+
+import java.io.File;
+import java.util.Optional;
+
+public class InsecureAuthenticatorConfig
```
Comment (kokosing) [inline presto-main/src/main/java/io/prestosql/server/security/InsecureAuthenticatorConfig.java:22]: add unit test

File: presto-main/src/main/java/io/prestosql/server/security/AuthenticationFilter.java [hunk 2]
```diff
@@ -41,14 +41,20 @@
     private final List<Authenticator> authenticators;
```
Comment (kokosing) [inline presto-main/src/main/java/io/prestosql/server/security/AuthenticationFilter.java:41]: can you please add some unit tests for this feature?

**Comment by findepi** [comment]
Let's have something in docs about this.

File: presto-main/src/main/java/io/prestosql/server/security/AuthenticationFilter.java (hunk 2 above)
Comment (dain) [reply presto-main/src/main/java/io/prestosql/server/security/AuthenticationFilter.java:41]: I updated the existing test in `TestResourceSecurity`

File: presto-main/src/main/java/io/prestosql/server/security/InsecureAuthenticator.java [hunk 3]
```diff
@@ -48,8 +60,14 @@ public Identity authenticate(ContainerRequestContext request)
             throw new AuthenticationException("Basic authentication or " + PRESTO_USER + " must be sent", BasicAuthCredentials.AUTHENTICATE_HEADER);
         }
 
-        return Identity.forUser(user)
-                .withPrincipal(new BasicPrincipal(user))
-                .build();
+        try {
+            String authenticatedUser = userMapping.mapUser(user);
+            return Identity.forUser(authenticatedUser)
+                    .withPrincipal(new BasicPrincipal(authenticatedUser))
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/server/security/InsecureAuthenticator.java:66]: Shouldn't this be the original user? That's what we do for `PasswordAuthenticator` (at least for the built-in ones, as a custom password authenticator can return any principal).

**Pull request merged**